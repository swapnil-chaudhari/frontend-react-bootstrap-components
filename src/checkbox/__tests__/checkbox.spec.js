import React from 'react';
import renderShallow from 'render-shallow';
import noop from 'src/utils/noop';
import Checkbox from '../checkbox';
import { findWithType } from 'react-shallow-testutils';

describe('<Checkbox>', () => {
    let component;

    describe('when it renders', () => {
        beforeAll(() => {
            const { output } = renderShallow(
                <Checkbox checked disabled onChange={ noop } />
            );

            component = output;
        });

        test('has proper markup', () => {
            expect(component).toMatchSnapshot();
        });
    });

    describe('it can add a partial state className', () => {
        beforeAll(() => {
            const { output } = renderShallow(
                <Checkbox partial checked disabled onChange={ noop } />
            );

            component = output;
        });

        test('has the className', () => {
            expect(findWithType(component, 'label').props.className).toEqual('partial');
        });
    });
});
