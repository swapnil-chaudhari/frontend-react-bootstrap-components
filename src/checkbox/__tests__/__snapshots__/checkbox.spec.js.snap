// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox> when it renders has proper markup 1`] = `
<div
  className="checkbox"
>
  <input
    checked={true}
    className="checkbox-input disabled"
    disabled={true}
    id="737884497096766615"
    onChange={[Function]}
    onClick={undefined}
    readOnly={undefined}
    type="checkbox"
  />
  <label
    className=""
    htmlFor="737884497096766615"
    onClick={undefined}
  />
</div>
`;
